test/framework/src/main/java/org/elasticsearch/index/mapper/MapperServiceTestCase.java

```diff
@@ -797,10 +801,13 @@ protected final String syntheticSource(DocumentMapper mapper, CheckedConsumer<XC
     private void roundTripSyntheticSource(DocumentMapper mapper, String syntheticSource, DirectoryReader reader) throws IOException {
         try (Directory roundTripDirectory = newDirectory()) {
             RandomIndexWriter roundTripIw = indexWriterForSyntheticSource(roundTripDirectory);
-            roundTripIw.addDocument(mapper.parse(new SourceToParse("1", new BytesArray(syntheticSource), XContentType.JSON)).rootDoc());
+            ParsedDocument doc = mapper.parse(new SourceToParse("1", new BytesArray(syntheticSource), XContentType.JSON));
+            doc.updateSeqID(0, 0);
+            doc.version().setLongValue(0);
+            roundTripIw.addDocuments(doc.docs());
```

Comment:
iiuc, this is to support multiple docs for nested. If so, let's include a short comment here to explain.

Reply:
https://github.com/elastic/elasticsearch/pull/109809/commits/2e3fef7bf86933da05d9c439988c40dc13acab39